python-backend/app/api/documents/mines/resources/expected_documents.py

```diff
@@ -36,7 +36,7 @@ def post(self, mine_guid):
             mine_exp_doc = MineExpectedDocument(
                 req_document_guid=new_doc['req_document_guid'],
                 exp_document_name=new_doc['document_name'],
-                exp_document_description=new_doc['document_description'],
+                exp_document_description=new_doc.get('document_description'),

```

Comment:
I'm assuming this is here because Document description is optional and you want a default.
From my understanding, .get() if it doesnt find anything returns NULL. I dont know what the requirement is, but you may want an empty string or something else as default. I'll leave that decision to you.

Also if you do prefer the .get() convention over direct accessors like ['key']. Then maybe we can have this as something to change as you go. I think we should change all our accessors to .get() as we shouldnt break the app everytime the dictionary doesnt have the required value. So if you can update that as you go, that'd be 💯  Starting possibly with this method change all accessors from ['key'] to .get('key', 'default value')

Reply:
I'm happy with null, also, the json str() casts the values when it sends it back from the api, so that would convert it to '' for the front end. 

exactly, .get is used because it's an optional field. are you thinking we should .get everywhere for consistency then validate fields that shouldn't be null with validators?

Reply:
I think we should change it everywhere so that if someone makes a call to the API using postman or whatever and doesn't put in one of the required fields. We can return an error we control, as opposed to the stack trace.

Reply:
Ok, let's make this a backlog item and try and get to it this sprint, Not in scope for this ticket? 

Does that work for you?